2.15.0beta crash - StudyOptionsFragment return from Activity launch

```
java.lang.RuntimeException: Unable to resume activity {com.ichi2.anki/com.ichi2.anki.DeckPicker}: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=1565212628, result=50, data=null} to activity {com.ichi2.anki/com.ichi2.anki.DeckPicker}: java.lang.NullPointerException: Attempt to invoke virtual method 'void androidx.appcompat.widget.Toolbar.setOnMenuItemClickListener(androidx.appcompat.widget.Toolbar$OnMenuItemClickListener)' on a null object reference
at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4626)
at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4659)
at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:52)
at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2261)
at android.os.Handler.dispatchMessage(Handler.java:107)
at android.os.Looper.loop(Looper.java:237)
at android.app.ActivityThread.main(ActivityThread.java:8107)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
Caused by: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=1565212628, result=50, data=null} to activity {com.ichi2.anki/com.ichi2.anki.DeckPicker}: java.lang.NullPointerException: Attempt to invoke virtual method 'void androidx.appcompat.widget.Toolbar.setOnMenuItemClickListener(androidx.appcompat.widget.Toolbar$OnMenuItemClickListener)' on a null object reference
at android.app.ActivityThread.deliverResults(ActivityThread.java:5324)
at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4613)
... 11 more
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void androidx.appcompat.widget.Toolbar.setOnMenuItemClickListener(androidx.appcompat.widget.Toolbar$OnMenuItemClickListener)' on a null object reference
at com.ichi2.anki.StudyOptionsFragment.configureToolbarInternal(StudyOptionsFragment.java:1)
at com.ichi2.anki.StudyOptionsFragment.configureToolbar(StudyOptionsFragment.java:1)
at com.ichi2.anki.StudyOptionsFragment.lambda$new$2(StudyOptionsFragment.java:1)
at com.ichi2.anki.StudyOptionsFragment.c(Unknown Source:0)
at com.ichi2.anki.f7.onActivityResult(Unknown Source:4)
at androidx.activity.result.ActivityResultRegistry.doDispatch(ActivityResultRegistry.java:3)
at androidx.activity.result.ActivityResultRegistry.dispatchResult(ActivityResultRegistry.java:3)
at androidx.activity.ComponentActivity.onActivityResult(ComponentActivity.java:1)
at androidx.fragment.app.FragmentActivity.onActivityResult(FragmentActivity.java:2)
at com.ichi2.anki.NavigationDrawerActivity.onActivityResult(NavigationDrawerActivity.java:9)
at com.ichi2.anki.DeckPicker.onActivityResult(DeckPicker.java:1)
at android.app.Activity.dispatchActivityResult(Activity.java:8294)
at android.app.ActivityThread.deliverResults(ActivityThread.java:5317)
```


```

--------- beginning of main
--
1 | 05-21 07:45:46.591 I/AnkiDroid(  481): AbstractFlashcardViewer:: EASE_2 pressed
2 | 05-21 07:45:46.604 I/AnkiDroid(  481): Answering card 1505428546031
3 | 05-21  07:45:46.668 I/AnkiDroid(  481): Rust: executeGetRowsAffected UPDATE  cards SET  factor=?,lapses=?,did=?,due=?,ivl=?,mod=?,usn=?,left=?,odid=?,odue=?,reps=?,type=?,queue=?  WHERE id = ?
4 | 05-21 07:45:46.694 I/AnkiDroid(  481): Obtaining card
5 | 05-21 07:45:46.723 I/AnkiDroid(  481): ViewGroupUtils::setRenderWorkaround - using default / hardware rendering
6 | 05-21 07:45:46.734 I/AnkiDroid(  481): AbstractFlashcardViewer:: Question successfully shown for card id 1505263840952
7 | 05-21 07:46:09.375 I/AnkiDroid(  481): AbstractFlashcardViewer:: Show answer button pressed
8 | 05-21 07:47:34.173 I/AnkiDroid(  481): AbstractFlashcardViewer:: EASE_2 pressed
9 | 05-21 07:47:34.177 I/AnkiDroid(  481): Answering card 1505263840952
10 | 05-21  07:47:34.188 I/AnkiDroid(  481): Rust: executeGetRowsAffected UPDATE  cards SET  factor=?,lapses=?,did=?,due=?,ivl=?,mod=?,usn=?,left=?,odid=?,odue=?,reps=?,type=?,queue=?  WHERE id = ?
11 | 05-21 07:47:34.203 I/AnkiDroid(  481): Obtaining card
12 | 05-21 07:47:34.233 I/AnkiDroid(  481): ViewGroupUtils::setRenderWorkaround - using default / hardware rendering
13 | 05-21 07:47:34.236 I/AnkiDroid(  481): AbstractFlashcardViewer:: Question successfully shown for card id 1505322876319
14 | 05-21 07:47:41.678 I/AnkiDroid(  481): AbstractFlashcardViewer:: Show answer button pressed
15 | 05-21 07:49:50.531 I/AnkiDroid(  481): AbstractFlashcardViewer:: EASE_2 pressed
16 | 05-21 07:49:50.533 I/AnkiDroid(  481): Answering card 1505322876319
17 | 05-21  07:49:50.543 I/AnkiDroid(  481): Rust: executeGetRowsAffected UPDATE  cards SET  factor=?,lapses=?,did=?,due=?,ivl=?,mod=?,usn=?,left=?,odid=?,odue=?,reps=?,type=?,queue=?  WHERE id = ?
18 | 05-21 07:49:50.594 I/AnkiDroid(  481): Obtaining card
19 | 05-21 07:49:50.612 I/AnkiDroid(  481): ViewGroupUtils::setRenderWorkaround - using default / hardware rendering
20 | 05-21 07:49:50.616 I/AnkiDroid(  481): AbstractFlashcardViewer:: Question successfully shown for card id 1496706787453
21 | 05-21 07:50:31.573 I/AnkiDroid(  481): AbstractFlashcardViewer:: Show answer button pressed
22 | 05-21 07:50:39.184 I/AnkiDroid(  481): Back key pressed
23 | 05-21 07:50:39.184 I/AnkiDroid(  481): finishWithAnimation END
24 | 05-21 07:50:39.196 I/AnkiDroid(  481): AnkiActivity::onPause - Reviewer
25 | 05-21 07:50:39.229 I/AnkiDroid(  481): AnkiActivity::onDestroy - DeckPicker
26 | 05-21 07:50:39.289 I/AnkiDroid(  481): AnkiActivity::onCreate - DeckPicker
27 | 05-21 07:50:39.299 I/AnkiDroid(  481): No startup screens required
28 | 05-21 07:50:39.304 I/AnkiDroid(  481): Triggering Automatic Sync
29 | 05-21 07:50:39.350 I/AnkiDroid(  481): Locked Collection - Collection Loading should fail
30 | 05-21 07:50:39.352 I/AnkiDroid(  481): Sync - starting sync
31 | 05-21 07:50:39.354 I/AnkiDroid(  481): ExecuteCommand: localMinutesWest
32 | 05-21 07:50:39.366 I/AnkiDroid(  481): Rust: executeGetRowsAffected UPDATE col SET dconf=?,decks=?
33 | 05-21 07:50:39.802 I/AnkiDroid(  481): flush - Saving information to DB...
34 | 05-21 07:50:39.803 I/AnkiDroid(  481): Rust: executeGetRowsAffected UPDATE col SET ls=?,crt=?,dty=?,mod=?,scm=?,usn=?,conf=?
35 | 05-21 07:50:40.155 I/AnkiDroid(  481): AnkiActivity::onStart - DeckPicker
36 | 05-21 07:50:40.164 I/AnkiDroid(  481): Handling Activity Result: 1565212628. Result: 50
37 | 05-21 07:50:40.165 I/AnkiDroid(  481): Creating notification channel with id/name: General Notifications/AnkiDroid
38 | 05-21 07:50:40.172 I/AnkiDroid(  481): Creating notification channel with id/name: Synchronization/Synchronization
39 | 05-21 07:50:40.173 I/AnkiDroid(  481): Creating notification channel with id/name: Global Reminders/Cards due
40 | 05-21 07:50:40.174 I/AnkiDroid(  481): Creating notification channel with id/name: Deck Reminders/Reminders
41 | --------- beginning of crash


```

It's the configureToolbar call in the activity launcher with the "RESULT_DEFAULT" return code. Why are view elements null when we come back from launching? Do we need to reinitialize views?

I'm going to instrument the code a bit and hope for analysis on the results

--- a/AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.java
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.java
@@ -163,6 +163,7 @@ public class StudyOptionsFragment extends Fragment implements Toolbar.OnMenuItem
     private void openFilteredDeckOptions(boolean defaultConfig) {
         Intent i = new Intent(getActivity(), FilteredDeckOptions.class);
         i.putExtra("defaultConfig", defaultConfig);
+        Timber.i("openFilteredDeckOptions()");
         mOnDeckOptionsActivityResult.launch(i);
         ActivityTransitionAnimation.slide(getActivity(), FADE);
     }
@@ -184,6 +185,7 @@ public class StudyOptionsFragment extends Fragment implements Toolbar.OnMenuItem
 
     @Override
     public void onCreate(@Nullable Bundle savedInstanceState) {
+        Timber.i("onCreate()");
         super.onCreate(savedInstanceState);
         //If we're being restored, don't launch deck options again.
         if (savedInstanceState == null && getArguments() != null) {
@@ -193,6 +195,7 @@ public class StudyOptionsFragment extends Fragment implements Toolbar.OnMenuItem
 
     @Override
     public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        Timber.i("onCreateView()");
         if (container == null) {
             // Currently in a layout without a container, so no reason to create our view.
             return null;
@@ -202,8 +205,8 @@ public class StudyOptionsFragment extends Fragment implements Toolbar.OnMenuItem
         mFragmented = getActivity().getClass() != StudyOptionsActivity.class;
         initAllContentViews(studyOptionsView);
         mToolbar = studyOptionsView.findViewById(R.id.studyOptionsToolbar);
-        mToolbar.inflateMenu(R.menu.study_options_fragment);
         if (mToolbar != null) {
+            mToolbar.inflateMenu(R.menu.study_options_fragment);
             configureToolbar();
         }
         refreshInterface(true);
@@ -217,14 +220,14 @@ public class StudyOptionsFragment extends Fragment implements Toolbar.OnMenuItem
         if (mFullNewCountThread != null) {
             mFullNewCountThread.interrupt();
         }
-        Timber.d("onDestroy()");
+        Timber.i("onDestroy()");
     }
 
 
     @Override
     public void onResume() {
         super.onResume();
-        Timber.d("onResume()");
+        Timber.i("onResume()");
         refreshInterface(true);
     }
 
@@ -244,10 +247,13 @@ public class StudyOptionsFragment extends Fragment implements Toolbar.OnMenuItem
 
 
     private void openReviewer() {
+        Timber.i("openReviewer()");
         Intent reviewer = new Intent(getActivity(), Reviewer.class);
         if (mFragmented) {
             mToReviewer = true;
+            Timber.i("openReviewer() fragmented mode");
             mOnRequestReviewActivityResult.launch(reviewer);
+            // TODO #8913 should we finish the activity here? when it comes back from review it's dead and mToolbar is null and it crashes
         } else {
             // Go to DeckPicker after studying when not tablet
             reviewer.setFlags(Intent.FLAG_ACTIVITY_FORWARD_RESULT);
@@ -328,6 +334,7 @@ public class StudyOptionsFragment extends Fragment implements Toolbar.OnMenuItem
                 openFilteredDeckOptions();
             } else {
                 Intent i = new Intent(getActivity(), DeckOptions.class);
+                Timber.i("Opening deck options for activity result");
                 mOnDeckOptionsActivityResult.launch(i);
                 ActivityTransitionAnimation.slide(getActivity(), FADE);
             }
@@ -374,6 +381,7 @@ public class StudyOptionsFragment extends Fragment implements Toolbar.OnMenuItem
     // This will allow a maximum of one recur in order to workaround database closes
     // caused by sync on startup where this might be running then have the collection close
     private void configureToolbarInternal(boolean recur) {
+        Timber.i("configureToolbarInternal()");
         try {
             mToolbar.setOnMenuItemClickListener(this);
             Menu menu = mToolbar.getMenu();
@@ -439,7 +447,12 @@ public class StudyOptionsFragment extends Fragment implements Toolbar.OnMenuItem
     }
 
     ActivityResultLauncher<Intent> mOnRequestReviewActivityResult = registerForActivityResult(new ActivityResultContracts.StartActivityForResult(), result -> {
-        configureToolbar();
+        Timber.i("StudyOptionsFragment::mOnRequestReviewActivityResult");
+        if (mToolbar != null) {
+            configureToolbar(); // FIXME we were crashing here because mToolbar is null #8913
+        } else {
+            AnkiDroidApp.sendExceptionReport("mToolbar null after return from tablet review session? Issue 8913", "StudyOptionsFragment");
+        }
         if ((result.getResultCode() == DeckPicker.RESULT_DB_ERROR) || (result.getResultCode() == DeckPicker.RESULT_MEDIA_EJECTED)) {
             closeStudyOptions(result.getResultCode());
             return;
@@ -454,6 +467,7 @@ public class StudyOptionsFragment extends Fragment implements Toolbar.OnMenuItem
     });
 
     ActivityResultLauncher<Intent> mOnDeckOptionsActivityResult = registerForActivityResult(new ActivityResultContracts.StartActivityForResult(), result -> {
+        Timber.i("StudyOptionsFragment::mOnDeckOptionsActivityResult");
         configureToolbar();
         if ((result.getResultCode() == DeckPicker.RESULT_DB_ERROR) || (result.getResultCode() == DeckPicker.RESULT_MEDIA_EJECTED)) {
             closeStudyOptions(result.getResultCode());

